src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpSender.php

```diff
@@ -47,20 +47,22 @@ public function send(Envelope $envelope): Envelope
         $delayStamp = $envelope->last(DelayStamp::class);
         $delay = $delayStamp ? $delayStamp->getDelay() : 0;
 
+        /** @var AmqpStamp|null $amqpStamp */
         $amqpStamp = $envelope->last(AmqpStamp::class);
         if (isset($encodedMessage['headers']['Content-Type'])) {
             $contentType = $encodedMessage['headers']['Content-Type'];
             unset($encodedMessage['headers']['Content-Type']);
 
-            $attributes = $amqpStamp ? $amqpStamp->getAttributes() : [];
-
-            if (!isset($attributes['content_type'])) {
-                $attributes['content_type'] = $contentType;
-
-                $amqpStamp = new AmqpStamp($amqpStamp ? $amqpStamp->getRoutingKey() : null, $amqpStamp ? $amqpStamp->getFlags() : AMQP_NOPARAM, $attributes);
+            if (!$amqpStamp || !isset($amqpStamp->getAttributes()['content_type'])) {
```

Comment:
This feels like the opposite of the merging logic in `Connection::publishOnExchnage()`. In that method, `headers` from the serializer win over headers from the `AmqpStamp`. But here, we're saying: "only use the `Content-Type` from the serializer if it is NOT already set on `AmqpStamp` (i.e. `AmqpStamp` wins)

Reply:
If people set the content-type manually on the envelope, we don't want to overwrite this. THis is how it has been before already.